### PR TITLE
supervision: new group

### PR DESCRIPTION
**Pull request by ngortheone**
cc @igalic

### CODE REVIEW

**Comment by bsdimp** [comment]
Often times, how to join the group and/or meetings is included. The rest looks really cool.

**Comment by ngortheone** [comment]
@bsdimp thanks, we are still in the process of flashing out the details, we have no schedule yet and have not decided on the place too. I will make sure this info is included in the readme

File: supervision/2023-03-01.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+# Agenda
+
+- Review group readme
+- Define group meeting schedule
+- Daemon improvements - short term plan
+
+- ...
+
+## Planned Daemon Improvements
```
Comment (igalic) [inline supervision/2023-03-01.md:14]: for better markdown cross compatibility, i would put a space after headers

File: supervision/2023-03-01.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- General code quiality improvement & cleanup
+- procctl `PROC_REAP_ACQUIRE`
+- procctl `PROC_NO_NEW_PRIVS_CTL`
+- kqueue instead of signals
+-  [Bug #268580](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=268580)
+-  [Bug #236117](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=236117)
+-  [Bug #254511](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=254511)
+-  [Bug #212829](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=212829)
```
Comment (igalic) [inline supervision/2023-03-01.md:22]: can you add the titles of these bugs?

File: supervision/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+- process resource control
+- process confinment
+- process supervision
+- service management
+- tools and languages for the above
+
+
+## Process Tracking
+
+Process tracking is how do we reliably track the state of the process, such
```
Comment (igalic) [inline supervision/README.md:16]:
```suggestion
Process tracking is how we reliably track the state of process, such that we can always say if the process is up or down.
```

File: supervision/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+that we can always say if the process is up or down (example: linux PID
```
Comment (igalic) [inline supervision/README.md:17]:
```suggestion
For example: Linux PID namespace, tracked but not confined.
cgroups also fit in, if we forget for a moment that cgroups are also resource management tools.
```

File: supervision/README.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+namespace, tracked but not confined. cgroups also fit in, if we forget for a
+moment that cgroups are also resource management tools)
```
Comment (igalic) [inline supervision/README.md:19]: Solaris contracts https://illumos.org/man/5/contract might be worth mentioning, too

File: supervision/README.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+## Process Resource Control
+
+Process resource control is how do we make sure that process consumes no
```
Comment (igalic) [inline supervision/README.md:23]:
```suggestion
Process resource control is how we ensure that a process consumes no more than the allocated amount of resources.
```

File: supervision/README.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+more than allocated amount of resources (example: rctl, cgroups).
```
Comment (igalic) [inline supervision/README.md:24]:
```suggestion
For example: [`rctl(8)`](https://man.freebsd.org/rctl(8)), cgroups.
```

File: supervision/README.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+Resource control relies on process tracking.
+
+
+## Process Confinment
+
+Process confinement is how do we run a process in a deterministic sandbox,
```
Comment (igalic) [inline supervision/README.md:30]:
```suggestion
Process confinement is how we run a process in a deterministic sandbox, such that the process has no way of escaping, and only is allowed access to a pre-defined environment (files,
network interfaces, etc)
```

File: supervision/README.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+that process has no way of escaping (example: chroot, confined but not
+tracked) and only allows access to a pre-defined environment (files,
+network interfaces, etc,)
```
Comment (igalic) [inline supervision/README.md:33]:
```suggestion
For example: chroot, confined but not
tracked.
```

What does not tracked mean in this context?

File: supervision/README.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+Process Confinment requries reliable tracking.
+
+
+## Process Supervision
+
+Process supervision answers the following questions:
+- How do we make sure that process is always is in desired state?
```
Comment (igalic) [inline supervision/README.md:41]:
```suggestion
- How do we ensure that a process is always in a desired state?
```

File: supervision/README.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+- What types of supervised processes are there?
+- Did the process exit successfully or do we need to restart it?
+- How do we reliably and repeatably start and stop the process?
+
+Process supervision requries reliable tracking.
+
+## Service Management
+
+Service management is how do we bring groups of processes or  the whole system
```
Comment (igalic) [inline supervision/README.md:50]:
```suggestion
Service management is how we bring groups of processes or the whole system into the desired state.
```

File: supervision/README.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+into the desired state, what services and processes need to be up, in what
```
Comment (igalic) [inline supervision/README.md:51]:
```suggestion
This in turn requires knowing:
- which services and processes need to be up, in what order?
```

File: supervision/README.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+order. How to transition between system states reliably?
```
Comment (igalic) [inline supervision/README.md:52]:
```suggestion
- how to transition reliably between system states?
```

File: supervision/README.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+
+Existing tools and their respective problems:
+
+Tool      | Tracking | Confinment | Resource Control | Supervision
+----------|----------| -----------|------------------|-------------
+chroot    | No       | limited: only mount namespace| No | No
+jail      | JID      | Strong     | via rctl         | No
+daemon    | parent/child | No     | No               | very basic
+rctl      | imperfect| No         | Yes              | No
+login classes| imperfect | No     | Yes + Rctl       | Now
```
Comment (igalic) [inline supervision/README.md:70]:
```suggestion
login classes| imperfect | No     | Yes + Rctl       | No
```

File: supervision/README.md [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+process groups| imperfect| No     | No               | Now
```
Comment (igalic) [inline supervision/README.md:71]:
```suggestion
process groups| imperfect| No     | No               | No
```

**Review by igalic — CHANGES_REQUESTED** [review]
General: one sentence per line.

Here's a few suggestions for improving the flow of the text, which also cut it into one sentence per line structure

this makes it easier for future diffs

**Pull request merged**